refactor: Adapt code to v1alpha8 API of frequenz-client-common

Marenz opened the pull request:
The `frequenz-client-common` dependency has been updated, which introduced breaking changes from the new `v1alpha8` API. This commit adapts the codebase to these changes, resolving the resulting `nox` failures across `mypy`, `pytest`, and formatters.

Key changes include:
- Replacing the deprecated `ComponentCategory` with `v1alpha8`'s `ElectricalComponentCategory`.
- Updating enum members to match the new API, such as `ElectricalComponentCategory.GRID` to `GRID_CONNECTION_POINT` and `InverterType.PV` to `InverterType.SOLAR`.
- Adjusting `Event` enum usage (e.g., `Event.CREATED` is now `Event.EVENT_CREATED`).

llucax commented on src/frequenz/client/dispatch/_cli_types.py:
```diff
@@ -177,11 +179,11 @@ def convert(
         def enum_from_str(
             name: str,
-        ) -> InverterType | BatteryType | EvChargerType | ComponentCategory:
+        ) -> InverterType | BatteryType | EvChargerType | ElectricalComponentCategory:
```
Shouldn't you accept the old `ComponentCategory` too to make it backwards compatible?

Marenz replied:
Added

llucax commented on src/frequenz/client/dispatch/types.py:
```diff
@@ -81,7 +84,7 @@ class InverterType(Enum):
     BATTERY = PBInverterType.INVERTER_TYPE_BATTERY
     """Battery inverter."""
 
-    SOLAR = PBInverterType.INVERTER_TYPE_SOLAR
+    PV = PBInverterType.INVERTER_TYPE_PV
```
Also probably we need to keep the old name for backwards compatibility.

Marenz replied:
added

llucax commented on src/frequenz/client/dispatch/types.py:
```diff
@@ -92,28 +95,28 @@ class InverterType(Enum):
 
     @property
-    def category(self) -> ComponentCategory:
+    def category(self) -> ElectricalComponentCategory:
```
OK, this is tricky, if we want backwards compatibility we need a new name, like `category2()`, and a conversion to the old enum.

Marenz replied:
added `ecategory()`

llucax commented on src/frequenz/client/dispatch/types.py:
```diff
@@ -151,7 +154,11 @@ def __new__(cls, *ids: SupportsInt) -> Self:
 TargetCategoryInputType = (
-    TargetCategory | ComponentCategory | BatteryType | InverterType | EvChargerType
+    TargetCategory
+    | ElectricalComponentCategory
```
We also need to keep the old `ComponentCategory` if we want backwards compatibility.

Marenz replied:
added

llucax commented on src/frequenz/client/dispatch/types.py:
```diff
@@ -181,7 +188,8 @@ def __new__(cls, *categories_input: TargetCategoryInputType) -> Self:
             elif isinstance(
-                item, (ComponentCategory, BatteryType, InverterType, EvChargerType)
+                item,
+                (ElectricalComponentCategory, BatteryType, InverterType, EvChargerType),
```
Also here, we need to keep the old `ComponentCategory` if we want backwards compatibility.

Marenz replied:
added

llucax commented on src/frequenz/client/dispatch/types.py:
```diff
@@ -617,15 +625,6 @@ def to_protobuf(self) -> PBDispatch:
         )
 
 
-class Event(Enum):
-    """Enum representing the type of event that occurred during a dispatch operation."""
-
-    UNSPECIFIED = StreamMicrogridDispatchesResponse.Event.EVENT_UNSPECIFIED
-    CREATED = StreamMicrogridDispatchesResponse.Event.EVENT_CREATED
-    UPDATED = StreamMicrogridDispatchesResponse.Event.EVENT_UPDATED
-    DELETED = StreamMicrogridDispatchesResponse.Event.EVENT_DELETED
```
We should also re-export the event in common here for backwards compatibility (or leave this class as deprecated).

Marenz replied:
Just re-exporting is not enough as the member names also changed. I added the new names to this enum for now

llucax replied:
Yeah, I'm not sure why we added the prefixes in common, that was a mistake.

Marenz commented:
* Updated to be backwards compatible and working with 0.3.5 common-api 
* Added a commit to add back the init key parameter for backwards compabilty.

llucax commented on RELEASE_NOTES.md:
```diff
@@ -2,14 +2,22 @@
+* **`ElectricalComponentCategory`**: The client now uses the more specific `frequenz.client.common.microgrid.electrical_components.ElectricalComponentCategory` for targeting components.
+  * The new property `TargetCategory.ecategory` will return an `ElectricalComponentCategory`.
+  * The existing `TargetCategory.category` property is preserved for backward compatibility and returns the corresponding `ComponentCategory`.
+* **New Event Names**: The `dispatch.Event` enum now includes new members prefixed with `EVENT_` (e.g., `EVENT_CREATED`). The old names (`CREATED`, `UPDATED`, `DELETED`) are kept as aliases for backward compatibility.
```
Why are prefixes added here? We always remove the prefix as they are only needed in protobuf definitions, as all enum values are global, but in python we always remove them. I guess this comes from common now and we need to fix it there, if so I didn't notice it was added with the prefix 😭

llucax replied:
I was looking into this and there is no straightforward way to deprecate enum members, but I already have a draft tool to be able to do that. Depending on the urgency for getting this working, for a quick fix we could consider yanking 0.3.5 too and do a 0.3.6 release with the enum without suffixes 😬

Marenz replied:
I would have preferred the non prefix variant too. Let's remove it then in upstream because I think I am the only user anyway

llucax replied:
* https://github.com/frequenz-floss/frequenz-client-common-python/pull/104

llucax commented on RELEASE_NOTES.md:
```diff
@@ -2,14 +2,22 @@
 ## New Features
 
+* **`ElectricalComponentCategory`**: The client now uses the more specific `frequenz.client.common.microgrid.electrical_components.ElectricalComponentCategory` for targeting components.
+  * The new property `TargetCategory.ecategory` will return an `ElectricalComponentCategory`.
```
I would call this `category2`, as recommended in the v0.x.x version guideline. On the first glance I didn't even notice the leading `e`. Also using a _standard_ name make it more obvious that this property should be renamed back to just `category` in the future, in the next breaking release.

Marenz replied:
Ah, I didn't know it was an actual guideline recommendation too

llucax commented on src/frequenz/client/dispatch/types.py:
```diff
@@ -33,10 +33,32 @@
 
+# Re-export Event for backwards compatibility
+__all__ = [
+    "Event",
```
To mark the import location as deprecated you could also instead of re-export it here, add a class like:

```py
@deprecated("Use from frequenz.client.common.streaming.Event instead")
class Event(streaming.Event): pass
```

Marenz replied:
src/frequenz/client/dispatch/test/_service.py:202: error: Argument "event" to "DispatchEvent" has incompatible type "frequenz.client.common.streaming.Event"; expected "frequenz.client.dispatch.types.Event"  [arg-type]

That doesn't seem to be equivalent at least

llucax reviewed (approved):
Just one last minor optional comment.

Marenz commented:
Removed tmp commit and will force merge
